node/src/components/block_executor.rs

```diff
@@ -83,7 +84,11 @@ type BlockHeight = u64;
 /// The Block executor component.
 #[derive(DataSize, Debug, Default)]
 pub(crate) struct BlockExecutor {
-    genesis_state_root_hash: Option<Digest>,
+    /// Height of the child of the highest known block at the time of initializing the component.
```

Comment:
I would like to see an explanation _why_ we need it.

Reply:
In order not to look for results of execution of blocks lower than that.

Previously when looking for pre-state hash, we were stopping at genesis child - now we stop at the child of the highest known block, and to do that, we need to know this block's height.

Also, we _have_ to do that, because the post-state hash of the highest known block might not match the pre-state hash of the next block, because an upgrade might have made changes to the global state in between.

Reply:
Ah, I meant in the code :D